File: micropython/button/message.py
from communication_protocol.communication_protocol import DeviceMessage
from communication_protocol.message_event import MessageEvent
from communication_protocol.message_type import MessageType


def on_click_request(mac: str) -> DeviceMessage:
    return DeviceMessage(
        MessageEvent.ON_CLICK,
        MessageType.REQUEST,
        mac,
        {},
    )


def on_hold_request(mac: str) -> DeviceMessage:
    return DeviceMessage(
        MessageEvent.ON_HOLD,
        MessageType.REQUEST,
        mac,
        {},
    )

File: arduino/button_esp8266_01_01M/src/CommunicationProtocol/message.cpp
#include "message.h"

DeviceMessage on_click_request(String mac) {
  JsonDocument payload;
  return DeviceMessage(String(millis()), "on_click", "request", mac, payload);
}

DeviceMessage on_hold_request(String mac) {
  JsonDocument payload;
  return DeviceMessage(String(millis()), "on_hold", "request", mac, payload);
}